halo2_proofs/Cargo.toml

```diff
@@ -62,11 +78,12 @@ rand_core = { version = "0.6", default-features = false, features = ["getrandom"
 getrandom = { version = "0.2", features = ["js"] }
 
 [features]
-default = ["shplonk"]
+default = ["shplonk", "gpu"]
```

Comment:
yeah， thanks very much.

Reply:
> I'm not sure about adding this as `default` now.
> 
> We will need some time to update the CI infra to support GPU usage. cc: @AronisAt79 @ntampakas @barryWhiteHat

hi @CPerezz.  
May I ask how long it will take for the CI infra to be ready？

In addition to update FFT data, we ported the multiexp operation [patch](https://github.com/filecoin-project/ec-gpu/compare/master...bchyl:ec-gpu:mexp-gpu?expand=1) compatible with pairing library based on  [filecoin ec-gpu](https://github.com/filecoin-project/ec-gpu) last week.  The performance data as following:
```
Testing Multiexp for 1024 elements...
GPU took 38ms.
CPU took 86ms.
Speedup: x2.2631578
============================
Testing Multiexp for 2048 elements...
GPU took 37ms.
CPU took 144ms.
Speedup: x3.891892
============================
Testing Multiexp for 4096 elements...
GPU took 38ms.
CPU took 270ms.
Speedup: x7.105263
============================
Testing Multiexp for 8192 elements...
GPU took 41ms.
CPU took 481ms.
Speedup: x11.731708
============================
Testing Multiexp for 16384 elements...
GPU took 43ms.
CPU took 847ms.
Speedup: x19.697674
============================
Testing Multiexp for 32768 elements...
GPU took 80ms.
CPU took 1608ms.
Speedup: x20.1
============================
Testing Multiexp for 65536 elements...
GPU took 63ms.
CPU took 3212ms.
Speedup: x50.984127
============================
```

Reply:
Hi @bchyl, some of these GPU times for FFT/MSM look great! For the MSM however it seems like you'll have to test for larger number of elements because the current numbers are very small and you don't yet see the expected ~linear increase in time you'd expect (so for small multiexps there are other overheads which makes sense but that doesn't really matter).

I do have a question about the CPU times for MSM and FFTs. They seem to be extremely slow for some reason, much slower than even when running them on a normal desktop CPU, and you're running them on a very powerful machine so that doesn't make any sense. On a standard machine (8 CPU cores) you can do an FFT of 2^20 in roughly 0.25s and an MSM of 2^20 in less than 10 seconds (the exact numbers of course dependent on the specific implementation). The numbers make it look like you're running only on a single core or something. Am I misinterpreting the data or do you think there may be something up with the CPU performance numbers?

Reply:
Hi @Brechtpd,  Thank you very much for your reference 8 cores performance data. 
We have double checked the data of our CPU (test machine is 80 cores) and it has been updated as mentioned above.

In general, for MSM it was been  consistent with your results,  2^20 less than 1s(864ms).  But for fft our result was 0.864s there was still much gap from your theoretical value 0,025s, we are debugging to fix it.

Reply:
Great, thanks for the updated numbers! I think the limited scalability beyond a certain number of CPU cores is at least partially caused by the multi-threading approach followed by the current CPU FFT/MSM implementations. Not really sure how big the impact is of that with a CPU with that many cores though.